src/skin/colorschemeparser.cpp

```diff
@@ -44,6 +47,11 @@ void ColorSchemeParser::setupLegacyColorSchemes(QDomElement docElem,
             WImageStore::setLoader(imsrc);
             WSkinColor::setLoader(imsrc);
 
+            // ronso0 :: find <SetVariable> nodes, update them.
+            // This calls SkinContext::updateVariables in skincontext.cpp which
+            // iterates over all <SetVariable> nodes
+            m_pContext->updateVariables;
```

Comment:
Looks good, just modify this method signature and pass the context to it.
When a new Scheme is selected from preferences the whole skin is rebuilt from scratch so we are sure this will be called and the new values set.